crates/ty_python_semantic/src/types/narrow.rs

```diff
@@ -235,7 +235,8 @@ impl ClassInfoConstraintFunction {
             | Type::TypeIs(_)
             | Type::WrapperDescriptor(_)
             | Type::DataclassTransformer(_)
-            | Type::TypedDict(_) => None,
+            | Type::TypedDict(_)
+            | Type::NewTypeInstance(_) => None,
```

Comment:
could we add an explicit test for this? e.g.

```py
from typing import NewType

N = NewType("N", int)

def f(x: N | str):
    # TODO: we should emit an error here (requires PEP-613 support)
    if isinstance(x, N):
        # we do not narrow the type here,
        # since the `isinstance()` call will actually fail at runtime!
        reveal_type(x)  # revealed: N | str
    else:
        reveal_type(x)  # revealed: N | str
```